R/template.R

```diff
@@ -266,133 +265,161 @@ prob_estimate_convert <- function(estimate) {
-#' `estimate` are of the same class, just supply a vector of length 1. If
-#' they are different, supply a vector of length 2. For matrices, it is best
-#' to supply `"numeric"` as the class to check here.
-#'
-#' @param estimator The type of averaging to use. By this point, the averaging
-#' type should be finalized, so this should be a character vector of length 1\.
-#' By default, this character value is required to be one of: `"binary"`,
-#' `"macro"`, `"micro"`, or `"macro_weighted"`. If your metric allows more
-#' or less averaging methods, override this with `averaging_override`.
-#'
-#' @param case_weights Optionally, the realized case weights, as a numeric
-#' vector. This must be the same length as `truth`, and will be considered in
-#' the `na_rm` checks. If supplied, this will be passed on to `metric_impl` as
-#' the named argument `case_weights`.
+#' @param case_weights the realized case weights, as a numeric vector. This must
+#'   be the same length as `truth`, and will be considered in the `na_rm`
+#'   checks. If supplied, this will be passed on to `metric_impl` as the named
+#'   argument `case_weights`.
 #'
 #' @param ... Extra arguments to your core metric function, `metric_impl`, can
-#' technically be passed here, but generally the extra args are added through
-#' R's scoping rules because the core metric function is created on the fly
-#' when the vector method is called.
+#'   technically be passed here, but generally the extra args are added through
+#'   R's scoping rules because the core metric function is created on the fly
+#'   when the vector method is called.
 #'
 #' @details
 #'
-#' `metric_vec_template()` is called from the vector implementation of your
-#' metric. Also defined inside your vector implementation is a separate
+#' `class_metric_vec_template()`, `numeric_metric_vec_template()`, and
+#' `prob_metric_vec_template()` are called from the vector implementation of
+#' your metric. Also defined inside your vector implementation is a separate
 #' function performing the core implementation of the metric function. This
-#' core function is passed along to `metric_vec_template()` as `metric_impl`.
+#' core function is passed along as `metric_impl`.
 #'
-#' @seealso [metric_summarizer()] [finalize_estimator()] [dots_to_estimate()]
+#' @seealso [metric-summarizers()] [finalize_estimator()] [dots_to_estimate()]
 #'
+#' @name metric-vec_template
+NULL
+
+#' @rdname metric-vec_template
 #' @export
-metric_vec_template <- function(metric_impl,
-                                truth,
-                                estimate,
-                                na_rm = TRUE,
-                                cls = "numeric",
-                                estimator = NULL,
-                                case_weights = NULL,
-                                ...) {
+numeric_metric_vec_template <- function(metric_impl,
+                                        truth,
```

Comment:
I'd also consider naming any new exported validation functions with a `check_*()` prefix. That is what we use in rlang, vctrs, dplyr, tidyr, etc, and is becoming the standard prefix for things like this